dts: edtlib: Improve error for wrong reg/ranges/interrupts size + add helper for testing error messages from edtlib

ulfalizer opened the pull request:
Show how the element size was calculated in the error message when a
'reg', 'ranges', or 'interrupts' property has the wrong size. This
should help with debugging. Also mention that #*-cells properties come
from the parent node, which can be subtle.

Came up in https://github.com/zephyrproject-rtos/zephyr/issues/21607
(though it seems the comment disappeared).

Also add a helper for testing error messages from `edtlib` (error messages from `dtlib` are already tested), and use it to verify the new hints:

```
dts: edtlib: Add helper for testing error messages and test _slice()

Add a helper function verify_error() to testedtlib.py that takes a DTS
source and verifies that it generates a particular EDTError.

Use it to test the hints that the previous commit added to errors
generated by _slice().
```

ulfalizer committed:
dts: edtlib: Improve error for wrong reg/ranges/interrupts size

Show how the element size was calculated in the error message when a
'reg', 'ranges', or 'interrupts' property has the wrong size. This
should help with debugging. Also mention that #*-cells properties come
from the parent node, which can be subtle.

Came up in https://github.com/zephyrproject-rtos/zephyr/issues/21607
(though it seems the comment disappeared).

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

pabigot reviewed (commented):
I really wish these sort of changes came with a change to the test that demonstrates and confirms that the output is as desired.

ulfalizer committed:
dts: edtlib: Add helper for testing error messages and test _slice()

Add a helper function verify_error() to testedtlib.py that takes a DTS
source and verifies that it generates a particular EDTError.

Use it to test the hints that the previous commit added to errors
generated by _slice().

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer commented:
@pabigot 
> I really wish these sort of changes came with a change to the test that demonstrates and confirms that the output is as desired.

Added a helper for testing error messages from `edtlib` and used it to test the new messages. See the new commit.
